lib/index.js

```diff
@@ -10,28 +10,44 @@ init = function(target) {
     Object.keys(target.rawAttributes).forEach(function (attr) {
       if (target.rawAttributes[attr].roles !== undefined) {
         if (target.rawAttributes[attr].roles === false) {
-          target.rawAttributes[attr].roles = {};
-          return; 
+          return;
         }
 
-        Object.keys(target.rawAttributes[attr].roles).forEach(function (role) {
-          if (typeof target.rawAttributes[attr].roles[role] === "boolean") {
-            target.rawAttributes[attr].roles[role] = {
-              set: target.rawAttributes[attr].roles[role],
-              get: target.rawAttributes[attr].roles[role]
-            };
+        Object.keys(target.rawAttributes[attr].roles).forEach(function (roleName) {
+          var role = target.rawAttributes[attr].roles[roleName];
+
+          if (roleName === 'default') {
+            if(typeof role === "string"){
+              role = target.rawAttributes[attr].roles.default = target.rawAttributes[attr].roles[role];
+            }
+          }
+
+          if (typeof role === "boolean") {
+            target.rawAttributes[attr].roles[roleName] = { set: role, get: role };
             return;
           }
 
-          if (target.rawAttributes[attr].roles[role].set === undefined) {
-            target.rawAttributes[attr].roles[role].set = false;
+          if (role.set === undefined) {
+            role.set = true;
           }
-          if (target.rawAttributes[attr].roles[role].get === undefined) {
-            target.rawAttributes[attr].roles[role].get = false;
+          if (role.get === undefined) {
+            role.get = true;
           }
         });
       }
     });
+    
+    var accessGranted = function(attr, options){
+      return !attr 
+        // If no role is given apply default if any
+        || !options.role && (typeof attr.roles !== 'object' || (attr.roles.default||{}).get !== false)
+        // If no roles defined in attribute or set to true
+        || attr.roles === undefined || attr.roles === true
+        // If role is given but not defined in attribute apply default if any
+        || options.role && (typeof attr.roles === 'object' && attr.roles[options.role] === undefined && (attr.roles.default||{}).get !== false)
```

Comment:
We should avoid configuration on call-site at all costs in favour of reasonable convention.

What i mean is that if you call `get('attr', {role: 'roleThatDoesNotExist'})` it should throw, not default to using the default role.